### ISSUE TITLE
no-use-before-define doesn't catch variable use before definition

### ISSUE THREAD

**Issue by SpadeAceman** [eslint/eslint#4831]
While refactoring some code, I got bitten by an issue I thought I'd told ESLint to catch for me.

```Javascript
/*eslint
	no-use-before-define:2
*/
;(function(){
	var x = y + 5 // this use of y before definition is caught
	var y = y + 5 // this use of y before definition is *not* caught
	var z = z // this use of z is also not caught, despite #4280
})
```
Output:
```
Error: "y" was used before it was defined (no-use-before-define), Line 5, Column 17
```

I've reviewed #4280, which I thought was intended to catch the errors on lines 6 and 7 as well, but it doesn't appear to have done so.

what | value
---- | ----
version | v1.10.3
rule | `no-use-before-define`
code | see above
expected | all three errors should be caught
actual | only the error on the first line is caught

**Comment by eslintbot** [comment]
@SpadeAceman Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by SpadeAceman** [comment]
Whoops, after doing some further digging I found a mention of this being resolved in `v2.0.0-alpha-1` in #4729.

**Comment by ilyavolodin** [comment]
That's correct. Those issues should be fixed in v2.0.0-alpha-1. Closing, feel free to reopen if the issue is still not fixed in alpha-1.
